static/js/metrics-explorer.js

```diff
@@ -70,16 +69,17 @@ async function metricsExplorerDatePickerHandler(evt) {
    
     Object.keys(queries).forEach(async function(queryName) {
         var queryDetails = queries[queryName];
-        await getQueryDetails(queryName, queryDetails);
+
         const tagsAndValue = await getTagKeyValue(queryDetails.metrics);
-        
-        queryDetails.everywhere = [];
-        queryDetails.everything = [];
         availableEverywhere = tagsAndValue.availableEverywhere.sort();
         availableEverything = tagsAndValue.availableEverything[0].sort();
         const queryElement = $(`.metrics-query .query-name:contains(${queryName})`).closest('.metrics-query');
         queryElement.find('.everywhere').autocomplete('option', 'source', availableEverywhere);
         queryElement.find('.everything').autocomplete('option', 'source', availableEverything);
+
+        queries[queryName] = queryDetails;
```

Comment:
This line is not needed, as in the code above, we are not changing the queryDetails object.